src/common/components/CalculatedExchangeAmount/CalculatedExchangeAmount.js

```diff
@@ -0,0 +1,66 @@
+import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+import {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import { AmountWithCurrencyField } from '@folio/stripes-acq-components';
+import { KeyValue } from '@folio/stripes/components';
+import { useStripes } from '@folio/stripes/core';
+
+import { useExchangeCalculation } from '../../hooks';
+
+const DEBOUNCE_DELAY = 500;
+
+export const CalculatedExchangeAmount = ({ currency, exchangeRate, total }) => {
+  const stripes = useStripes();
+  const systemCurrency = stripes.currency;
+  const enabled = Boolean(systemCurrency !== currency && total);
+
+  const [exchangeProps, setExchangeProps] = useState({
```

Comment:
this state, debouncedSave and useEffect look strange, what do you solve with it?

in query key you don't use object, but it's values - so you don't need to preserve previous state

Reply:
The main purpose is to prevent multiple requests via `debounce` function

Reply:
`const { exchangedAmount } = useExchangeCalculation({ to: systemCurrency, amount: +total, etc }, { enabled });`

it's the same with what you did, but without extra strange code

Reply:
there is a chance you'll need to update useQuery config a bit to achieve this, but no need to do what you've done

Reply:
----
now I see why it's required - form changes values, at the same time all this logic should be moved to `useExchangeCalculation`, component shouldn't know anything about it